core/Pos/Core/Common/Coin.hs

```diff
@@ -50,7 +50,7 @@ unsafeAddCoin :: Coin -> Coin -> Coin
 unsafeAddCoin (unsafeGetCoin -> a) (unsafeGetCoin -> b)
     | res >= a && res >= b && res <= unsafeGetCoin (maxBound @Coin) = mkCoin res
     | otherwise =
-      -- TODO [CSL-2173]: Clarify
+      -- This function is unsafe
```

Comment:
>We're basically doing compiler's job here (with something like LiquidHaskell we wouldn't have to), so let's be as formal as possible.

That's not an argument. You see 'addCoin' function and "it can panic" keyword, you read the code => voila. No need for duplicating the check condition twice.